Changed Asset images to System images

FrankElias77 opened the pull request:
Fixes #17 
### Summary
Start the app with the current screen instead of the intro screen

#### Core Tests
Tested on Manticore, generic_hmi, SYNC 3.0

Commit:
Asset images to system images

Commit:
Code update

Commit:
Code update

joeljfischer commented on MobileWeather/ImageProcessor.m:
```diff
@@ -31,9 +31,25 @@ - (instancetype)init {
+- (UIImage *)imageByName:(NSString *) imageName {
+    UIImage *imageReturned;
+
+    if (@available(iOS 13.0, *)) {
```
Move the app to iOS 13.0+ and remove these checks and the internal images

joeljfischer commented on MobileWeather/ImageProcessor.m:
```diff
@@ -31,9 +31,25 @@ - (instancetype)init {
     return self;
 }
 
+- (UIImage *)imageByName:(NSString *) imageName {
```
```suggestion
- (UIImage *)imageByName:(NSString *)imageName {
```

Based on moving the app to iOS 13.0+, I don't think this method will be necessary anymore?

joeljfischer commented on MobileWeather/ImageProcessor.h:
```diff
@@ -11,6 +11,8 @@
 
 + (ImageProcessor *)sharedProcessor;
 
+- (UIImage *)imageByName:(NSString *) imageName;
```
```suggestion
- (UIImage *)imageByName:(NSString *)imageName;
```

joeljfischer commented on MobileWeather/ImageProcessor.m:
```diff
@@ -48,4 +64,50 @@ - (NSData *)dataFromConditionImage:(NSString *)conditionImage {
 
+- (NSString *)systemImageFromAssetsImage:(NSString *) imageName {
+    NSString *newImageName = imageName;
+    if ([imageName isEqual:@"chancerain"]) {
```
```suggestion
    if ([imageName isEqualToString:@"chancerain"]) {
```

All of these would need to be `isEqualToString`

joeljfischer commented on MobileWeather/ImageProcessor.m:
```diff
@@ -48,4 +64,50 @@ - (NSData *)dataFromConditionImage:(NSString *)conditionImage {
     return data;
 }
 
+- (NSString *)systemImageFromAssetsImage:(NSString *) imageName {
```
```suggestion
- (NSString *)mw_systemImageFromAssetsImage:(NSString *)imageName {
```

Commit:
Comments update

joeljfischer commented on SmartDeviceLinkService.m:
```diff
@@ -78,8 +78,8 @@ - (void)start {
     // Change which config you need based on if you want to connect to a TDK (default) or a wifi based emulator (debug)
-    SDLLifecycleConfiguration *lifecycleConfig = [SDLLifecycleConfiguration defaultConfigurationWithAppName:@"MobileWeather" fullAppId:@"330533107"];
-//    SDLLifecycleConfiguration *lifecycleConfig = [SDLLifecycleConfiguration debugConfigurationWithAppName:@"MobileWeather" fullAppId:@"330533107" ipAddress:@"192.168.45.2" port:12345];
+//    SDLLifecycleConfiguration *lifecycleConfig = [SDLLifecycleConfiguration defaultConfigurationWithAppName:@"MobileWeather" fullAppId:@"330533107"];
+    SDLLifecycleConfiguration *lifecycleConfig = [SDLLifecycleConfiguration debugConfigurationWithAppName:@"MobileWeather" fullAppId:@"330533107" ipAddress:@"m.sdl.tools" port:12957];
```
This needs to be default by...default

joeljfischer replied:
Still needs to be fixed

joeljfischer commented on MobileWeather/ImageProcessor.m:
```diff
@@ -32,8 +32,8 @@ - (instancetype)init {
 }
 
 - (UIImage *)imageFromConditionImage:(NSString *)conditionImage {
```
To fix the blurriness, you should add a parameter here for an enum of "image size", large (primary graphic) == 256 (very crisp on TDK), small (for soft buttons / menu / choice set) == 64? Maybe 128? Try it out and see what works. You can also pass a weight (smaller items should be weighted thicker?)

Below comment describes what to do with these parameters.

joeljfischer commented on MobileWeather/ImageProcessor.m:
```diff
@@ -32,8 +32,8 @@ - (instancetype)init {
 - (UIImage *)imageFromConditionImage:(NSString *)conditionImage {
-    UIImage *image = [UIImage imageNamed:conditionImage];
-    
+    UIImage *image = [UIImage systemImageNamed:[self mw_systemImageFromAssetsImage:conditionImage]];
```
```suggestion
    UIImage *image = [UIImage systemImageNamed:[self mw_systemImageFromAssetsImage:conditionImage] withConfiguration:[UIImageSymbolConfiguration configurationWithPointSize:256 weight:UIImageSymbolWeightBold]];
```

This is an example of 256 points, Bold

Commit:
Comments review

joeljfischer commented on MobileWeather/ImageSize.h:
```diff
@@ -0,0 +1,18 @@
+typedef CGFloat ImageSize;
+
+extern ImageSize const ImageSizeLargeGraphic_256;
+extern ImageSize const ImageSizeSmallGraphic_64;
```
Use an enum, not a constant.

```objc
typedef NS_ENUM(NSUInteger, ImageSize) {
    ImageSizeLarge,
    ImageSizeSmall
};

@interface ImageSizeHelper : NSObject

+ (CGFloat)floatForImageSize:(ImageSize)imageSize;

@end
```

joeljfischer commented on MobileWeather/ImageSize.m:
```diff
@@ -0,0 +1,12 @@
+
+#import "ImageSize.h"
+
+ImageSize const ImageSizeLargeGraphic_256 = 256;
+ImageSize const ImageSizeSmallGraphic_64 = 64;
```
Implement the changes from .h

joeljfischer commented on MobileWeather/ImageProcessor.h:
```diff
@@ -6,12 +6,13 @@
 + (ImageProcessor *)sharedProcessor;
 
-- (UIImage *)imageFromConditionImage:(NSString *)conditionImage;
+- (UIImage *)imageFromConditionImage:(NSString *)conditionImage imageSize:(ImageSize)imageSize;
```
Maybe add an `SDLArtwork *` helper method like `- (SDLArtwork *)artworkFromConditionImage`

Commit:
ImageSize update

Commit:
ImageProcessor update

artworkFromConditionImage

joeljfischer committed:
Merge branch 'develop' into feature/issue-17-replace-internal-images-with-ios-system-images

# Conflicts:
#	SmartDeviceLinkService.m

joeljfischer commented on MobileWeather/ImageProcessor.m:
```diff
@@ -31,9 +31,13 @@ - (instancetype)init {
+}
+
+- (UIImage *)imageFromConditionImage:(NSString *)conditionImage imageSize:(ImageSize)imageSize {
+    UIImage *image = [UIImage systemImageNamed:[self mw_systemImageFromAssetsImage:conditionImage] withConfiguration:[UIImageSymbolConfiguration configurationWithPointSize:[ImageSizeHelper floatForImageSize:imageSize] weight:UIImageSymbolWeightBold]];
```
This should probably include being a template image instead of the SDLArtwork method?

joeljfischer replied:
```suggestion
    UIImage *image = [UIImage systemImageNamed:[self mw_systemImageFromAssetsImage:conditionImage] withConfiguration:[UIImageSymbolConfiguration configurationWithPointSize:[ImageSizeHelper floatForImageSize:imageSize] weight:UIImageSymbolWeightMedium]];
```

joeljfischer commented on MobileWeather/ImageProcessor.m:
```diff
@@ -31,9 +31,13 @@ - (instancetype)init {
-- (UIImage *)imageFromConditionImage:(NSString *)conditionImage {
-    UIImage *image = [UIImage imageNamed:conditionImage];
-    
+- (SDLArtwork *)artworkFromConditionImage:(NSString *)conditionImage imageSize:(ImageSize)imageSize {
+    return [SDLArtwork artworkWithImage:[[self imageFromConditionImage:conditionImage imageSize:imageSize] imageWithRenderingMode:UIImageRenderingModeAlwaysTemplate] asImageFormat:SDLArtworkImageFormatPNG];
+}
```
The ordering should follow the .h file, so this should be below the `imageFromConditionImage` method

joeljfischer commented on SmartDeviceLinkService.m:
```diff
@@ -605,7 +605,7 @@ - (void)repeatWeatherInformation {
 - (NSArray<SDLSoftButtonObject *> *)buildDefaultSoftButtons {
     __weak typeof(self) weakSelf = self;
-    SDLSoftButtonState *currentWeatherState = [[SDLSoftButtonState alloc] initWithStateName:@"state" text:self.localization[@"sb.current"] image:[[UIImage imageNamed:@"clear-day"] imageWithRenderingMode:UIImageRenderingModeAlwaysTemplate]];
+    SDLSoftButtonState *currentWeatherState = [[SDLSoftButtonState alloc] initWithStateName:@"state" text:self.localization[@"sb.current"] image:[[UIImage systemImageNamed:@"sun.max" withConfiguration:[UIImageSymbolConfiguration configurationWithPointSize:[ImageSizeHelper floatForImageSize:ImageSizeSmall] weight:UIImageSymbolWeightBold]] imageWithRenderingMode:UIImageRenderingModeAlwaysTemplate]] ...
```
Use the artwork initializer for the soft button state and the ImageProcessor to make this much shorter and simpler. Same for everywhere else.

joeljfischer commented:
None of the above resolved comments were done. Did you forget to push?

Commit:
ImageProcessor update

artworkFromConditionImage update

joeljfischer commented on MobileWeather/ImageProcessor.h:
```diff
@@ -6,12 +6,16 @@
-- (UIImage *)imageFromConditionImage:(NSString *)conditionImage;
+- (UIImage *)imageFromConditionImage:(NSString *)conditionImage imageSize:(ImageSize)imageSize;
+
+- (SDLArtwork *)artworkFromConditionImage:(NSString *)conditionImage imageSize:(ImageSize)imageSize isPersistent:(BOOL)isPersistent isTemplate:(BOOL)isTemplate;
```
I think everything is `isTemplate = YES`, so you can remove this option and make it always YES in the method body.

joeljfischer replied:
```suggestion
- (SDLArtwork *)artworkFromConditionImage:(NSString *)conditionImage imageSize:(ImageSize)imageSize isPersistent:(BOOL)isPersistent;
```

joeljfischer commented on MobileWeather/ImageProcessor.m:
```diff
@@ -31,21 +31,77 @@ - (instancetype)init {
+        artwork = [[SDLArtwork alloc] initWithImage:[[self imageFromConditionImage:conditionImage imageSize:imageSize] imageWithRenderingMode:UIImageRenderingModeAlwaysTemplate] persistent:isPersistent asImageFormat:SDLArtworkImageFormatPNG];
+    } else {
+        artwork = [[SDLArtwork alloc] initWithImage:[self imageFromConditionImage:conditionImage imageSize:imageSize] persistent:isPersistent asImageFormat:SDLArtworkImageFormatPNG];
+    }
```
See above comment

joeljfischer commented on SmartDeviceLinkService.m:
```diff
@@ -78,8 +78,8 @@ - (void)start {
-    SDLLifecycleConfiguration *lifecycleConfig = [SDLLifecycleConfiguration defaultConfigurationWithAppName:@"MobileWeather" fullAppId:@"330533107"];
-//    SDLLifecycleConfiguration *lifecycleConfig = [SDLLifecycleConfiguration debugConfigurationWithAppName:@"MobileWeather" fullAppId:@"330533107" ipAddress:@"192.168.45.2" port:12345];
+//    SDLLifecycleConfiguration *lifecycleConfig = [SDLLifecycleConfiguration defaultConfigurationWithAppName:@"MobileWeather" fullAppId:@"330533107"];
+    SDLLifecycleConfiguration *lifecycleConfig = [SDLLifecycleConfiguration debugConfigurationWithAppName:@"MobileWeather" fullAppId:@"330533107" ipAddress:@"192.168.45.2" port:12345];
```
This needs to be changed back

joeljfischer commented on SmartDeviceLinkService.m:
```diff
@@ -605,7 +605,7 @@ - (void)repeatWeatherInformation {
 - (NSArray<SDLSoftButtonObject *> *)buildDefaultSoftButtons {
     __weak typeof(self) weakSelf = self;
-    SDLSoftButtonState *currentWeatherState = [[SDLSoftButtonState alloc] initWithStateName:@"state" text:self.localization[@"sb.current"] image:[[UIImage imageNamed:@"clear-day"] imageWithRenderingMode:UIImageRenderingModeAlwaysTemplate]];
+    SDLSoftButtonState *currentWeatherState = [[SDLSoftButtonState alloc] initWithStateName:@"state" text:self.localization[@"sb.current"] artwork:[[ImageProcessor sharedProcessor] artworkFromConditionImage:@"clear-day" imageSize:ImageSizeSmall isPersistent:NO isTemplate:YES]];
```
All soft buttons should be persistent

joeljfischer commented on MobileWeather/MainViewController.m:
```diff
@@ -105,7 +105,7 @@ - (void)setWeatherConditions:(WeatherConditions *)conditions withLanguage:(Weath
     self.statusLabel.text = status;
     self.currentConditionsLabel.text = conditions.conditionTitle;
-    self.conditionIcon.image = [[ImageProcessor sharedProcessor] imageFromConditionImage:conditions.conditionIcon];
+    self.conditionIcon.image = [[ImageProcessor sharedProcessor] imageFromConditionImage:conditions.conditionIcon imageSize:[ImageSizeHelper floatForImageSize:ImageSizeLarge]];
```
I think the image size here isn't correct?

joeljfischer reviewed (changes requested):
Just a few more items

Commit:
Comments review

joeljfischer commented on MobileWeather/ImageProcessor.m:
```diff
@@ -31,21 +31,74 @@ - (instancetype)init {
 }
 
+- (SDLArtwork *)artworkFromConditionImage:(NSString *)conditionImage imageSize:(ImageSize)imageSize isPersistent:(BOOL)isPersistent isTemplate:(BOOL)isTemplate {
+    SDLArtwork *artwork;
+    artwork = [[SDLArtwork alloc] initWithImage:[[self imageFromConditionImage:conditionImage imageSize:imageSize] imageWithRenderingMode:UIImageRenderingModeAlwaysTemplate] persistent:isPersistent asImageFormat:SDLArtworkImageFormatPNG];
+
+    return artwork;
```
```suggestion
    return [[SDLArtwork alloc] initWithImage:[[self imageFromConditionImage:conditionImage imageSize:imageSize] imageWithRenderingMode:UIImageRenderingModeAlwaysTemplate] persistent:isPersistent asImageFormat:SDLArtworkImageFormatPNG];;
```

joeljfischer commented on MobileWeather/MainViewController.m:
```diff
@@ -105,7 +105,7 @@ - (void)setWeatherConditions:(WeatherConditions *)conditions withLanguage:(Weath
     self.statusLabel.text = status;
     self.currentConditionsLabel.text = conditions.conditionTitle;
-    self.conditionIcon.image = [[ImageProcessor sharedProcessor] imageFromConditionImage:conditions.conditionIcon];
+    self.conditionIcon.image = [[ImageProcessor sharedProcessor] imageFromConditionImage:conditions.conditionIcon imageSize:[ImageSizeHelper floatForImageSize:ImageSizeSmall]];
```
```suggestion
    self.conditionIcon.image = [[ImageProcessor sharedProcessor] imageFromConditionImage:conditions.conditionIcon imageSize:ImageSizeSmall];
```

I think this is still wrong?

Commit:
Image processor update

Commit:
artworkFromConditionImage update

Commit:
main.storyboard update

Updated the build deployment target for storyboard to match the project based on warning in xcode.

joeljfischer commented on MobileWeather/ImageProcessor.m:
```diff
@@ -31,21 +31,72 @@ - (instancetype)init {
 }
 
+- (SDLArtwork *)artworkFromConditionImage:(NSString *)conditionImage imageSize:(ImageSize)imageSize isPersistent:(BOOL)isPersistent {
+    
```
```suggestion
```

joeljfischer commented on MobileWeather/ImageProcessor.h:
```diff
@@ -6,12 +6,16 @@
+
+- (SDLArtwork *)artworkFromConditionImage:(NSString *)conditionImage imageSize:(ImageSize)imageSize isPersistent:(BOOL)isPersistent;
 
 - (NSData *)dataFromConditionImage:(NSString *)conditionImage;
```
This method is not used. Delete it here and the .m

joeljfischer commented on MobileWeather/MainViewController.m:
```diff
@@ -105,7 +107,7 @@ - (void)setWeatherConditions:(WeatherConditions *)conditions withLanguage:(Weath
     self.statusLabel.text = status;
     self.currentConditionsLabel.text = conditions.conditionTitle;
-    self.conditionIcon.image = [[ImageProcessor sharedProcessor] imageFromConditionImage:conditions.conditionIcon];
+    self.conditionIcon.image = [[ImageProcessor sharedProcessor] imageFromConditionImage:conditions.conditionIcon imageSize:ImageSizeSmall];
```
```suggestion
    self.conditionIcon.image = [[ImageProcessor sharedProcessor] imageFromConditionImage:conditions.conditionIcon imageSize:ImageSizeLarge];
```

joeljfischer commented on MobileWeather/MainViewController.m:
```diff
@@ -48,6 +48,8 @@ - (void)viewDidLoad {
     
     self.versionLabel.text = [NSString stringWithFormat:@"%@ (%@)", bundleAppVersion, bundleBuildVersion];
 
+    self.conditionIcon.contentMode = UIViewContentModeScaleAspectFill;
```
You can change this in the storyboard by clicking on the image view, then changing "Content Mode" here:
![Screen Shot 2021-07-28 at 8 57 42 AM](https://user-images.githubusercontent.com/1767063/127325889-646ed737-4d7e-4a90-a15c-4ce6815d6554.png)

Commit:
conditionImage content mode update

Commit:
unused method update

Commit:
conditionIcon size edit
